src/polynomial.jl

```diff
@@ -0,0 +1,65 @@
+function Polynomial{C,E}(coefficients::AbstractVector, exponents::AbstractMatrix) where {C <: VecOrMat_eltypes, E <: Union{CxxWrap.CxxLong, Int64}}
+    v = convert(Vector{C}, coefficients)
+    m = convert(Matrix{E}, exponents)
+    return Polynomial{to_cxx_type(C),to_cxx_type(E)}(v, m)
+end
+
+function Polynomial{C,E}(p::Polynomial) where {C,E}
+    v = coefficients_as_vector(p)
+    m = monomials_as_matrix(p)
+    return Polynomial{C,E}(v, m)
+end
+
+function Polynomial{C,E}(c::Number, nvars::Base.Integer) where {C,E}
+    m = Matrix{E}(undef,1,nvars)
+    return Polynomial{C,E}([c],m)
+end
+
+# deriving template type from input
+Polynomial(vec::AbstractVector{C}, mat::AbstractMatrix{E}) where {C,E} =
+    Polynomial{promote_to_pm_type(Vector,C),promote_to_pm_type(Matrix,E)}(vec, mat)
+Polynomial{C}(vec::AbstractVector, mat::AbstractMatrix{E}) where {C,E} =
+    Polynomial{C,promote_to_pm_type(Matrix,E)}(vec, mat)
+
+# defaulting to {Rational,Int64}
+Polynomial(x...) = Polynomial{Rational,Int64}(x)
+Polynomial{C}(x...) where C = Polynomial{Rational,Int64}(x)
```

Comment:
`Polynomial{C}(x...) where C = Polynomial{C,Int64}(x)` ?

Reply:
yeah, it shold be that